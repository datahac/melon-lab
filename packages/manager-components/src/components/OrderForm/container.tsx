import { withFormik } from 'formik';
import { compose, withHandlers } from 'recompose';
import * as Yup from 'yup';
import {
  divide,
  greaterThan,
  max,
  min,
  multiply,
} from '../../utils/functionalBigNumber';
import OrderForm from './index';

const calculateInputs = (props, field, value) => {
  const { values, info } = props;
  let maxTotal;
  let maxQuantity;

  const typeValue = field === 'type' ? value : values.orderType;
  const totalValue = field === 'total' ? value : values.total;
  const quantityValue = field === 'quantity' ? value : values.quantity;

  if (values.strategy === 'Market') {
    maxTotal =
      typeValue === 'Buy'
        ? min(info.tokens.quoteToken.balance, totalValue)
        : totalValue;
    maxQuantity =
      typeValue === 'Sell'
        ? max(info.tokens.baseToken.balance, quantityValue)
        : quantityValue;
  } else if (values.strategy === 'Limit') {
    maxTotal = typeValue === 'Buy' ? info.tokens.quoteToken.balance : Infinity;
    maxQuantity =
      typeValue === 'Sell' ? info.tokens.baseToken.balance : Infinity;
  }

  if (field === 'total') {
    if (greaterThan(value, maxTotal)) {
      props.setFieldValue('total', maxTotal);
    } else if (values.price) {
      const quantity = divide(value, values.price);
      if (values.quantity !== value) {
        props.setFieldValue('quantity', quantity);
      }
    }
  }

  if (field === 'quantity') {
    if (greaterThan(value, maxQuantity)) {
      props.setFieldValue('quantity', maxQuantity);
    } else if (values.price) {
      const total = multiply(value, values.price);
      if (values.total !== value) {
        props.setFieldValue('total', total);
      }
    }
  }

  if (field === 'price' && values.quantity) {
    props.setFieldValue('total', multiply(values.quantity, value));
  }
};

const validation = props => {
  const numberFormat = (0).toFixed(props.decimals);
  const minNumber = numberFormat.slice(0, -1) + '1';

  return Yup.object().shape({
    price: Yup.number()
      .min(minNumber, `Minimum price is ${minNumber}`)
      .required('Price is required.'),
    quantity: Yup.number()
      .min(minNumber, `Minimum quantity is ${minNumber}`)
      .required('Quantity is required.'),
    total: Yup.number()
      .min(minNumber, `Minimum total is ${minNumber}`)
      .required('Total is required.'),
  });
};

const withFormValidation = withFormik({
  mapPropsToValues: props => ({ ...props.values }),
  validationSchema: props => validation(props),
  enableReinitialize: true,
  handleSubmit: (values, form) => {
    if (form.props.onSubmit) {
      form.props.onSubmit(values);
    }
  },
});

const withFormHandler = compose(
  withHandlers({
    onChange: props => event => {
      props.setFieldValue(event.target.name, event.target.value);
      calculateInputs(props, event.target.name, event.target.value);
    },
  }),
);

export default compose(
  withFormValidation,
  withFormHandler,
)(OrderForm);
