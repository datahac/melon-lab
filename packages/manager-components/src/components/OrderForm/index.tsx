import React, { StatelessComponent } from 'react';
import Button from '~/blocks/Button';
// import Dropdown from '~/blocks/Dropdown';
import Form from '~/blocks/Form';
import Input from '~/blocks/Input';
import Notification from '~/blocks/Notification';
import Switch from '~/blocks/Switch';
import Toggle from '~/blocks/Toggle';
import OrderInfo from '~/components/OrderInfo';

import styles from './styles.css';

interface FormValues {
  price: string;
  quantity: string;
  total: string;
  exchange: string;
  orderType: string;
  strategy: string;
}

export interface OrderFormProps {
  values: FormValues;
  handleSubmit?: () => void;
  handleBlur?: () => void;
  onChange?: React.ChangeEvent<any>;
  info?: any;
  baseTokenSymbol?: string;
  quoteTokenSymbol?: string;
  exchanges: Array<{
    name: string;
    label: string;
  }>;
  selectedOrder?: () => void;
  errors: any;
  touched: any;
  decimals?: number;
  type?: string;
  dataValid?: boolean;
  isManager?: boolean;
  isCompetition?: boolean;
}

export const OrderForm: StatelessComponent<OrderFormProps> = ({
  handleSubmit,
  handleBlur,
  onChange,
  info,
  baseTokenSymbol,
  quoteTokenSymbol,
  exchanges,
  selectedOrder,
  errors,
  values,
  touched,
  decimals,
  dataValid,
  isManager,
  isCompetition,
}) => {
  const isMarket = values.strategy === 'Market' ? true : false;
  const numberPlaceholder = (0).toFixed(decimals);

  return (
    <div className="order-form">
      <style jsx>{styles}</style>
      <h3>Trade</h3>
      <Form>
        {!dataValid && (
          <Notification isWarning>
            Trading not possible when price feed down
          </Notification>
        )}
        <div className="order-form__toggles">
          <div className="order-form__toggle">
            <Toggle
              name="strategy"
              value="Market"
              text="Market"
              isChecked={values.strategy === 'Market'}
              onChange={onChange}
            />
          </div>
          <div className="order-form__toggle">
            <Toggle
              name="strategy"
              value="Limit"
              text="Limit"
              isChecked={values.strategy === 'Limit'}
              onChange={onChange}
            />
          </div>
        </div>

        <div className="order-form__switch">
          <Switch
            options={[baseTokenSymbol, quoteTokenSymbol]}
            labels={['Buy', 'Sell']}
            onChange={onChange}
            name="orderType"
            value={values.orderType}
            isChecked={values.orderType === 'Sell' ? true : false}
            disabled={isMarket || !dataValid || !isManager}
          />
        </div>
        {/* <div className="order-form__dropdown">
          <Dropdown
            name="exchange"
            value={values.exchange}
            options={exchanges}
            label="Exchange"
            onChange={onChange}
            disabled={isMarket || !dataValid}
          />
        </div> */}
        <div className="order-form__order-info">
          <OrderInfo {...info} />
        </div>
        <div className="order-form__input">
          <Input
            value={values.price}
            disabled={isMarket || !dataValid || !isManager}
            type="number"
            label="Price"
            name="price"
            insideLabel="true"
            placeholder={numberPlaceholder}
            decimals={decimals}
            onChange={onChange}
            onBlur={handleBlur}
            required={true}
            formatNumber={true}
            error={touched.price && errors.price}
          />
        </div>
        <div className="order-form__input">
          <Input
            value={values.quantity}
            type="number"
            label="Quantity"
            name="quantity"
            insideLabel="true"
            placeholder={numberPlaceholder}
            decimals={decimals}
            onChange={onChange}
            onBlur={handleBlur}
            required={true}
            formatNumber={true}
            error={touched.quantity && errors.quantity}
            disabled={(isMarket && !values.price) || !dataValid || !isManager}
          />
        </div>
        <div className="order-form__input">
          <Input
            value={values.total}
            type="number"
            label="Total"
            name="total"
            insideLabel="true"
            placeholder={numberPlaceholder}
            decimals={decimals}
            onChange={onChange}
            onBlur={handleBlur}
            required={true}
            formatNumber={true}
            error={touched.total && errors.total}
            disabled={(isMarket && !values.price) || !dataValid || !isManager}
          />
        </div>
        <Button
          disabled={(isMarket && !values.price) || !dataValid || !isManager}
          onClick={handleSubmit}
          type="submit"
        >
          {values.orderType}
        </Button>
      </Form>
    </div>
  );
};

export default OrderForm;
