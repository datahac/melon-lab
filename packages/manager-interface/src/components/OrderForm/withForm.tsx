import { withFormik } from 'formik';
import * as Yup from 'yup';
import {
  divide,
  greaterThan,
  max,
  min,
  multiply,
} from '~/utils/functionalBigNumber';
import { withHandlers } from 'recompose';

const formCalculation = (props, field, value) => {
  const { values, info } = props;
  let maxTotal;
  let maxQuantity;

  const typeValue = field === 'type' ? value : values.orderType;
  const totalValue = field === 'total' ? value : values.total;
  const quantityValue = field === 'quantity' ? value : values.quantity;

  if (values.strategy === 'Market') {
    maxTotal =
      typeValue === 'Buy'
        ? min(info.tokens.quoteToken.balance, totalValue)
        : totalValue;
    maxQuantity =
      typeValue === 'Sell'
        ? max(info.tokens.baseToken.balance, quantityValue)
        : quantityValue;
  } else if (values.strategy === 'Limit') {
    maxTotal = typeValue === 'Buy' ? info.tokens.quoteToken.balance : Infinity;
    maxQuantity =
      typeValue === 'Sell' ? info.tokens.baseToken.balance : Infinity;
  }

  if (field === 'total') {
    if (greaterThan(value, maxTotal)) {
      props.setFieldValue('total', maxTotal);
    } else if (values.price) {
      const quantity = divide(value, values.price).toString(10);
      if (values.quantity !== value) {
        props.setFieldValue('quantity', quantity);
      }
    }
  }

  if (field === 'quantity') {
    if (greaterThan(value, maxQuantity)) {
      props.setFieldValue('quantity', maxQuantity);
    } else if (values.price) {
      const total = multiply(value, values.price).toString(10);
      if (values.total !== value) {
        props.setFieldValue('total', total);
      }
    }
  }

  if (field === 'price' && values.quantity) {
    const total = multiply(values.quantity, value).toString(10);
    props.setFieldValue('total', total);
  }
};

const validation = props => {
  const numberFormat = (0).toFixed(props.decimals || 4);
  const minNumber = numberFormat.slice(0, -1) + '1';

  return Yup.object().shape({
    price: Yup.number()
      .min(minNumber, `Minimum price is ${minNumber}`)
      .required('Price is required.'),
    quantity: Yup.number()
      .min(minNumber, `Minimum quantity is ${minNumber}`)
      .required('Quantity is required.'),
    total: Yup.number()
      .min(minNumber, `Minimum total is ${minNumber}`)
      .required('Total is required.'),
  });
};

const initialValues = {
  price: '',
  type: 'sell',
  strategy: 'Market',
  quantity: '',
  total: '',
  exchange: '',
};

const withForm = withFormik({
  mapPropsToValues: props =>
    props.formValues ? { ...props.formValues } : initialValues,
  validationSchema: props => validation(props),
  enableReinitialize: true,
  handleSubmit: (values, form) =>
    form.props.onSubmit && form.props.onSubmit(values),
});

const withFormHandlers = withHandlers({
  onChange: props => event => {
    const { setFieldValue } = props;
    const { name, value } = event.target;

    setFieldValue(name, value);
    formCalculation(props, name, value);
  },
});

export { withForm, withFormHandlers };
