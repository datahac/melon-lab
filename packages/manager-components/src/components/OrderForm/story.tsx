import { storiesOf } from '@storybook/react';
import React from 'react';
import OrderForm from './index';

const data = {
  baseTokenSymbol: 'ETH-T-M',
  quoteTokenSymbol: 'MLN-T-M',
  selectedOrder: false,
  isManager: true,
  info: {
    lastPrice: 0.5,
    bid: 0,
    ask: 0,
    tokens: {
      baseToken: {
        name: 'ETH-T',
        balance: 30.0,
      },
      quoteToken: {
        name: 'MLN-T',
        balance: 20.0,
      },
    },
  },
  exchanges: [
    { value: 'RADAR_RELAY', name: 'Radar Relay' },
    { value: 'OASIS_DEX', name: 'OasisDEX' },
  ],
  values: {
    price: '1.0000',
    orderType: 'Sell',
    strategy: 'Limit',
    quantity: '1.0000',
    total: '1.0000',
    exchange: 'OASIS_DEX',
  },
  touched: {
    price: null,
    orderType: null,
    strategy: null,
    quantity: null,
    total: null,
    exchange: null,
  },
  error: {
    price: null,
    orderType: null,
    strategy: null,
    quantity: null,
    total: null,
    exchange: null,
  },
  selectedExchange: 'RadarRelay',
  selectedOrderType: 'Buy',
  decimals: 4,
  priceFeedUp: true,
};

storiesOf('Components|Order Form', module).add('Default', () => {
  return <OrderForm {...data} />;
});
