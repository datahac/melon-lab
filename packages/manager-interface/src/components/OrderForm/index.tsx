import React from 'react';
import Composer from 'react-composer';
import OrderForm from '~/components/OrderForm';
import { NetworkConsumer } from '+/components/NetworkContext';
import { FundManagerConsumer } from '+/components/FundManagerContext';
import { withForm, withFormHandlers } from './withForm';
import isSameAddress from '~/utils/isSameAddress';
import { compose } from 'recompose';

const WrappedOrderForm = compose(
  withForm,
  withFormHandlers,
)(OrderForm);

export default class OrderFormContainer extends React.PureComponent {
  getTokenBalance = asset => {
    const { holdings } = this.props;
    return {
      name: asset,
      balance:
        holdings &&
        holdings.length &&
        holdings.find(holding => holding.symbol === asset).balance.toString(10),
    };
  };

  render() {
    return (
      <Composer components={[<NetworkConsumer />, <FundManagerConsumer />]}>
        {([network, associatedFund]) => {
          const {
            address,
            quoteAsset,
            baseAsset,
            holdings,
            formValues,
          } = this.props;

          const isManager =
            !!associatedFund && isSameAddress(associatedFund, address);

          const tokens = {
            baseToken: this.getTokenBalance(this.props.baseAsset),
            quoteToken: this.getTokenBalance(this.props.quoteAsset),
          };

          return (
            <WrappedOrderForm
              tokens={tokens}
              isCompetition={false}
              isManager={isManager}
              holdings={holdings}
              quoteAsset={quoteAsset}
              baseAsset={baseAsset}
              formValues={formValues}
              priceFeedUp={network && network.priceFeedUp}
            />
          );
        }}
      </Composer>
    );
  }
}
