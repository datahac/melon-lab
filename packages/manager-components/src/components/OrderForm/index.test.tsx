import React from 'react';
import OrderForm from './container';

const data = {
  baseTokenSymbol: 'ETH-T-M',
  quoteTokenSymbol: 'MLN-T-M',
  strategy: 'Limit',
  info: {
    lastPrice: 0.5,
    bid: 0,
    ask: 0,
    tokens: {
      baseToken: {
        name: 'ETH-T',
        balance: 30,
      },
      quoteToken: {
        name: 'MLN-T',
        balance: 20,
      },
    },
  },
  exchanges: [
    { value: 'RadarRelay', name: 'Radar Relay' },
    { value: 'OasisDEX', name: 'OasisDEX' },
  ],
  selectedExchange: 'RadarRelay',
  selectedOrderType: 'Buy',
  decimals: 6,
  dataValid: true,
  total: '',
};

describe('OrderForm', () => {
  const defaultElement = <OrderForm {...data} />;

  it('should render correctly', () => {
    const wrapper = shallow(defaultElement);
    expect(wrapper).toMatchSnapshot();
  });
});
