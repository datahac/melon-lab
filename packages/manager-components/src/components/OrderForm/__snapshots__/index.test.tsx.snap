// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderForm should render correctly 1`] = `
<WithFormik(withHandlers(Component))
  baseTokenSymbol="ETH-T-M"
  dataValid={true}
  decimals={6}
  exchanges={
    Array [
      Object {
        "name": "Radar Relay",
        "value": "RadarRelay",
      },
      Object {
        "name": "OasisDEX",
        "value": "OasisDEX",
      },
    ]
  }
  info={
    Object {
      "ask": 0,
      "bid": 0,
      "lastPrice": 0.5,
      "tokens": Object {
        "baseToken": Object {
          "balance": 30,
          "name": "ETH-T",
        },
        "quoteToken": Object {
          "balance": 20,
          "name": "MLN-T",
        },
      },
    }
  }
  quoteTokenSymbol="MLN-T-M"
  selectedExchange="RadarRelay"
  selectedOrderType="Buy"
  strategy="Limit"
  total=""
>
  <Formik
    baseTokenSymbol="ETH-T-M"
    dataValid={true}
    decimals={6}
    enableReinitialize={true}
    exchanges={
      Array [
        Object {
          "name": "Radar Relay",
          "value": "RadarRelay",
        },
        Object {
          "name": "OasisDEX",
          "value": "OasisDEX",
        },
      ]
    }
    handleSubmit={[Function]}
    info={
      Object {
        "ask": 0,
        "bid": 0,
        "lastPrice": 0.5,
        "tokens": Object {
          "baseToken": Object {
            "balance": 30,
            "name": "ETH-T",
          },
          "quoteToken": Object {
            "balance": 20,
            "name": "MLN-T",
          },
        },
      }
    }
    initialValues={Object {}}
    isInitialValid={false}
    onSubmit={[Function]}
    quoteTokenSymbol="MLN-T-M"
    render={[Function]}
    selectedExchange="RadarRelay"
    selectedOrderType="Buy"
    strategy="Limit"
    total=""
    validateOnBlur={true}
    validateOnChange={true}
    validationSchema={[Function]}
  >
    <withHandlers(Component)
      baseTokenSymbol="ETH-T-M"
      dataValid={true}
      decimals={6}
      dirty={false}
      errors={Object {}}
      exchanges={
        Array [
          Object {
            "name": "Radar Relay",
            "value": "RadarRelay",
          },
          Object {
            "name": "OasisDEX",
            "value": "OasisDEX",
          },
        ]
      }
      handleBlur={[Function]}
      handleChange={[Function]}
      handleReset={[Function]}
      handleSubmit={[Function]}
      info={
        Object {
          "ask": 0,
          "bid": 0,
          "lastPrice": 0.5,
          "tokens": Object {
            "baseToken": Object {
              "balance": 30,
              "name": "ETH-T",
            },
            "quoteToken": Object {
              "balance": 20,
              "name": "MLN-T",
            },
          },
        }
      }
      initialValues={Object {}}
      isSubmitting={false}
      isValid={false}
      isValidating={false}
      quoteTokenSymbol="MLN-T-M"
      registerField={[Function]}
      resetForm={[Function]}
      selectedExchange="RadarRelay"
      selectedOrderType="Buy"
      setError={[Function]}
      setErrors={[Function]}
      setFieldError={[Function]}
      setFieldTouched={[Function]}
      setFieldValue={[Function]}
      setFormikState={[Function]}
      setStatus={[Function]}
      setSubmitting={[Function]}
      setTouched={[Function]}
      setValues={[Function]}
      strategy="Limit"
      submitCount={0}
      submitForm={[Function]}
      total=""
      touched={Object {}}
      unregisterField={[Function]}
      validateField={[Function]}
      validateForm={[Function]}
      validateOnBlur={true}
      validateOnChange={true}
      values={Object {}}
    >
      <Component
        baseTokenSymbol="ETH-T-M"
        dataValid={true}
        decimals={6}
        dirty={false}
        errors={Object {}}
        exchanges={
          Array [
            Object {
              "name": "Radar Relay",
              "value": "RadarRelay",
            },
            Object {
              "name": "OasisDEX",
              "value": "OasisDEX",
            },
          ]
        }
        handleBlur={[Function]}
        handleChange={[Function]}
        handleReset={[Function]}
        handleSubmit={[Function]}
        info={
          Object {
            "ask": 0,
            "bid": 0,
            "lastPrice": 0.5,
            "tokens": Object {
              "baseToken": Object {
                "balance": 30,
                "name": "ETH-T",
              },
              "quoteToken": Object {
                "balance": 20,
                "name": "MLN-T",
              },
            },
          }
        }
        initialValues={Object {}}
        isSubmitting={false}
        isValid={false}
        isValidating={false}
        onChange={[Function]}
        quoteTokenSymbol="MLN-T-M"
        registerField={[Function]}
        resetForm={[Function]}
        selectedExchange="RadarRelay"
        selectedOrderType="Buy"
        setError={[Function]}
        setErrors={[Function]}
        setFieldError={[Function]}
        setFieldTouched={[Function]}
        setFieldValue={[Function]}
        setFormikState={[Function]}
        setStatus={[Function]}
        setSubmitting={[Function]}
        setTouched={[Function]}
        setValues={[Function]}
        strategy="Limit"
        submitCount={0}
        submitForm={[Function]}
        total=""
        touched={Object {}}
        unregisterField={[Function]}
        validateField={[Function]}
        validateForm={[Function]}
        validateOnBlur={true}
        validateOnChange={true}
        values={Object {}}
      >
        <div
          className="jsx-__scopedHash order-form"
        >
          <JSXStyle
            css="__scoped"
            styleId="__scopedHash"
          />
          <h3
            className="jsx-__scopedHash"
          >
            Trade
          </h3>
          <Form
            onSubmit={[Function]}
          >
            <form
              className="form"
              onSubmit={[Function]}
            >
              <div
                className="jsx-__scopedHash order-form__toggles"
              >
                <div
                  className="jsx-__scopedHash order-form__toggle"
                >
                  <Toggle
                    isChecked={false}
                    name="strategy"
                    onChange={[Function]}
                    text="Market"
                    value="Market"
                  >
                    <label
                      className="jsx-__scopedHash toggle"
                    >
                      <JSXStyle
                        css="__scoped"
                        styleId="__scopedHash"
                      />
                      <input
                        checked={false}
                        className="jsx-__scopedHash toggle__input"
                        hidden={true}
                        name="strategy"
                        onChange={[Function]}
                        type="radio"
                        value="Market"
                      />
                      <span
                        className="jsx-__scopedHash toggle__text"
                      >
                        Market
                      </span>
                    </label>
                  </Toggle>
                </div>
                <div
                  className="jsx-__scopedHash order-form__toggle"
                >
                  <Toggle
                    isChecked={false}
                    name="strategy"
                    onChange={[Function]}
                    text="Limit"
                    value="Limit"
                  >
                    <label
                      className="jsx-__scopedHash toggle"
                    >
                      <JSXStyle
                        css="__scoped"
                        styleId="__scopedHash"
                      />
                      <input
                        checked={false}
                        className="jsx-__scopedHash toggle__input"
                        hidden={true}
                        name="strategy"
                        onChange={[Function]}
                        type="radio"
                        value="Limit"
                      />
                      <span
                        className="jsx-__scopedHash toggle__text"
                      >
                        Limit
                      </span>
                    </label>
                  </Toggle>
                </div>
              </div>
              <div
                className="jsx-__scopedHash order-form__switch"
              >
                <Switch
                  disabled={true}
                  isChecked={false}
                  labels={
                    Array [
                      "Buy",
                      "Sell",
                    ]
                  }
                  name="orderType"
                  onChange={[Function]}
                  options={
                    Array [
                      "ETH-T-M",
                      "MLN-T-M",
                    ]
                  }
                >
                  <div
                    className="jsx-__scopedHash switch"
                  >
                    <JSXStyle
                      css="__scoped"
                      styleId="__scopedHash"
                    />
                    <label
                      className="jsx-__scopedHash switch__wrapper"
                    >
                      <input
                        checked={false}
                        className="jsx-__scopedHash switch__input"
                        disabled={true}
                        name="orderType"
                        onChange={[Function]}
                        type="checkbox"
                      />
                      <span
                        className="jsx-__scopedHash switch__label"
                        data-label-left="Buy"
                        data-label-right="Sell"
                      >
                        ETH-T-M
                      </span>
                      <span
                        className="jsx-__scopedHash switch__label"
                        data-label-left="Buy"
                        data-label-right="Sell"
                      >
                        MLN-T-M
                      </span>
                      <div
                        className="jsx-__scopedHash switch__icon"
                      />
                    </label>
                  </div>
                </Switch>
              </div>
              <div
                className="jsx-__scopedHash order-form__order-info"
              >
                <OrderInfo
                  ask={0}
                  bid={0}
                  lastPrice={0.5}
                  tokens={
                    Object {
                      "baseToken": Object {
                        "balance": 30,
                        "name": "ETH-T",
                      },
                      "quoteToken": Object {
                        "balance": 20,
                        "name": "MLN-T",
                      },
                    }
                  }
                >
                  <div
                    className="jsx-__scopedHash order-info"
                  >
                    <JSXStyle
                      css="__scoped"
                      styleId="__scopedHash"
                    />
                    <div
                      className="jsx-__scopedHash order-info__prices"
                    >
                      <div
                        className="jsx-__scopedHash order-info__last-price"
                      >
                        <NumberFormat
                          allowEmptyFormatting={false}
                          allowNegative={true}
                          decimalScale={4}
                          decimalSeparator="."
                          displayType="text"
                          fixedDecimalScale={true}
                          getInputRef={[Function]}
                          isAllowed={[Function]}
                          isNumericString={false}
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onMouseUp={[Function]}
                          onValueChange={[Function]}
                          prefix=""
                          suffix=""
                          thousandSeparator={true}
                          type="text"
                          value={0.5}
                        >
                          <span>
                            0.5000
                          </span>
                        </NumberFormat>
                        <span
                          className="jsx-__scopedHash order-info__price-desc"
                        >
                          Last Price
                        </span>
                      </div>
                      <div
                        className="jsx-__scopedHash order-info__bid"
                      >
                        <span
                          className="jsx-__scopedHash"
                        >
                          N/A
                        </span>
                        <span
                          className="jsx-__scopedHash order-info__price-desc"
                        >
                          Bid
                        </span>
                      </div>
                      <div
                        className="jsx-__scopedHash order-info__ask"
                      >
                        <span
                          className="jsx-__scopedHash"
                        >
                          N/A
                        </span>
                        <span
                          className="jsx-__scopedHash order-info__price-desc"
                        >
                          Ask
                        </span>
                      </div>
                    </div>
                    <div
                      className="jsx-__scopedHash order-info__balances"
                    >
                      <div
                        className="jsx-__scopedHash order-info__balance"
                      >
                        <span
                          className="jsx-__scopedHash order-info__balance-desc"
                        >
                          ETH-T
                          :
                        </span>
                         
                        <NumberFormat
                          allowEmptyFormatting={false}
                          allowNegative={true}
                          decimalScale={4}
                          decimalSeparator="."
                          displayType="text"
                          fixedDecimalScale={true}
                          getInputRef={[Function]}
                          isAllowed={[Function]}
                          isNumericString={false}
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onMouseUp={[Function]}
                          onValueChange={[Function]}
                          prefix=""
                          suffix=""
                          thousandSeparator={true}
                          type="text"
                          value={30}
                        >
                          <span>
                            30.0000
                          </span>
                        </NumberFormat>
                      </div>
                      <div
                        className="jsx-__scopedHash order-info__balance"
                      >
                        <span
                          className="jsx-__scopedHash order-info__balance-desc"
                        >
                          MLN-T
                          :
                        </span>
                         
                        <NumberFormat
                          allowEmptyFormatting={false}
                          allowNegative={true}
                          decimalScale={4}
                          decimalSeparator="."
                          displayType="text"
                          fixedDecimalScale={true}
                          getInputRef={[Function]}
                          isAllowed={[Function]}
                          isNumericString={false}
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onMouseUp={[Function]}
                          onValueChange={[Function]}
                          prefix=""
                          suffix=""
                          thousandSeparator={true}
                          type="text"
                          value={20}
                        >
                          <span>
                            20.0000
                          </span>
                        </NumberFormat>
                      </div>
                    </div>
                  </div>
                </OrderInfo>
              </div>
              <div
                className="jsx-__scopedHash order-form__input"
              >
                <Input
                  decimals={6}
                  disabled={true}
                  formatNumber={true}
                  insideLabel="true"
                  label="Price"
                  name="price"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="0.000000"
                  required={true}
                >
                  <div
                    className="jsx-__scopedHash input input--inside-label"
                  >
                    <JSXStyle
                      css="__scoped"
                      styleId="__scopedHash"
                    />
                    <span
                      className="jsx-__scopedHash input__label"
                    >
                      Price
                    </span>
                    <NumberFormat
                      allowEmptyFormatting={false}
                      allowNegative={true}
                      className="input__field"
                      decimalScale={6}
                      decimalSeparator="."
                      disabled={true}
                      displayType="input"
                      fixedDecimalScale={true}
                      getInputRef={[Function]}
                      isAllowed={[Function]}
                      isNumericString={true}
                      name="price"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onMouseUp={[Function]}
                      onValueChange={[Function]}
                      placeholder="0.000000"
                      prefix=""
                      suffix=""
                      type="text"
                    >
                      <input
                        className="input__field"
                        disabled={true}
                        name="price"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onMouseUp={[Function]}
                        placeholder="0.000000"
                        type="text"
                        value=""
                      />
                    </NumberFormat>
                    <div
                      className="jsx-__scopedHash input__error"
                    />
                  </div>
                </Input>
              </div>
              <div
                className="jsx-__scopedHash order-form__input"
              >
                <Input
                  decimals={6}
                  disabled={true}
                  formatNumber={true}
                  insideLabel="true"
                  label="Quantity"
                  name="quantity"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="0.000000"
                  required={true}
                >
                  <div
                    className="jsx-__scopedHash input input--inside-label"
                  >
                    <JSXStyle
                      css="__scoped"
                      styleId="__scopedHash"
                    />
                    <span
                      className="jsx-__scopedHash input__label"
                    >
                      Quantity
                    </span>
                    <NumberFormat
                      allowEmptyFormatting={false}
                      allowNegative={true}
                      className="input__field"
                      decimalScale={6}
                      decimalSeparator="."
                      disabled={true}
                      displayType="input"
                      fixedDecimalScale={true}
                      getInputRef={[Function]}
                      isAllowed={[Function]}
                      isNumericString={true}
                      name="quantity"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onMouseUp={[Function]}
                      onValueChange={[Function]}
                      placeholder="0.000000"
                      prefix=""
                      suffix=""
                      type="text"
                    >
                      <input
                        className="input__field"
                        disabled={true}
                        name="quantity"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onMouseUp={[Function]}
                        placeholder="0.000000"
                        type="text"
                        value=""
                      />
                    </NumberFormat>
                    <div
                      className="jsx-__scopedHash input__error"
                    />
                  </div>
                </Input>
              </div>
              <div
                className="jsx-__scopedHash order-form__input"
              >
                <Input
                  decimals={6}
                  disabled={true}
                  formatNumber={true}
                  insideLabel="true"
                  label="Total"
                  name="total"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="0.000000"
                  required={true}
                >
                  <div
                    className="jsx-__scopedHash input input--inside-label"
                  >
                    <JSXStyle
                      css="__scoped"
                      styleId="__scopedHash"
                    />
                    <span
                      className="jsx-__scopedHash input__label"
                    >
                      Total
                    </span>
                    <NumberFormat
                      allowEmptyFormatting={false}
                      allowNegative={true}
                      className="input__field"
                      decimalScale={6}
                      decimalSeparator="."
                      disabled={true}
                      displayType="input"
                      fixedDecimalScale={true}
                      getInputRef={[Function]}
                      isAllowed={[Function]}
                      isNumericString={true}
                      name="total"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onMouseUp={[Function]}
                      onValueChange={[Function]}
                      placeholder="0.000000"
                      prefix=""
                      suffix=""
                      type="text"
                    >
                      <input
                        className="input__field"
                        disabled={true}
                        name="total"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onMouseUp={[Function]}
                        placeholder="0.000000"
                        type="text"
                        value=""
                      />
                    </NumberFormat>
                    <div
                      className="jsx-__scopedHash input__error"
                    />
                  </div>
                </Input>
              </div>
              <Button
                disabled={true}
                type="submit"
              >
                <button
                  className="jsx-__scopedHash button button--medium button--primary"
                  disabled={true}
                  onClick={[Function]}
                  type="submit"
                >
                  <JSXStyle
                    css="__scoped"
                    styleId="__scopedHash"
                  />
                </button>
              </Button>
            </form>
          </Form>
        </div>
      </Component>
    </withHandlers(Component)>
  </Formik>
</WithFormik(withHandlers(Component))>
`;
